src/aio/native_linux_aio_provider.cpp

```diff
@@ -26,107 +26,119 @@
 
 #include "native_linux_aio_provider.h"
 
-#include <errno.h>
-#include <fcntl.h>
-#include <unistd.h>
-#include <memory>
-
 #include "aio/aio_provider.h"
 #include "aio/disk_engine.h"
+#include "rocksdb/env.h"
+#include "rocksdb/slice.h"
+#include "rocksdb/status.h"
 #include "runtime/service_engine.h"
 #include "runtime/task/async_calls.h"
+#include "utils/env.h"
 #include "utils/fmt_logging.h"
 #include "utils/latency_tracer.h"
 #include "utils/ports.h"
-#include "utils/safe_strerror_posix.h"
 
 namespace dsn {
 
 native_linux_aio_provider::native_linux_aio_provider(disk_engine *disk) : aio_provider(disk) {}
 
 native_linux_aio_provider::~native_linux_aio_provider() {}
 
-linux_fd_t native_linux_aio_provider::open(const char *file_name, int flag, int pmode)
+std::unique_ptr<rocksdb::RandomAccessFile>
+native_linux_aio_provider::open_read_file(const std::string &fname)
 {
-    auto fd = ::open(file_name, flag, pmode);
-    if (fd == DSN_INVALID_FILE_HANDLE) {
-        LOG_ERROR("create file '{}' failed, err = {}", file_name, utils::safe_strerror(errno));
+    std::unique_ptr<rocksdb::RandomAccessFile> rfile;
+    auto s = dsn::utils::PegasusEnv(dsn::utils::FileDataType::kSensitive)
```

Comment:
means all files would be read or written as encrypted files ?

Reply:
Yes. We can improve to distinguish files which are not necessary to encrypt, however, it's not a problem if all files are encrypted.